packages/core/helper-plugin/src/content-manager/CMEditViewDataManager.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+
+import { MessageDescriptor } from 'react-intl';
+
+import type { Attribute, Schema } from '@strapi/strapi';
+
+/**
+ * TODO: All these types could be accurately inferred from
+ * their usage in the content-manager when we move it back.
+ */
+interface CMEditViewDataManagerContext {
+  addComponentToDynamicZone?: (
+    keys: string[],
+    componentLayoutData: Record<string, unknown>,
+    allComponents: Record<string, unknown>,
+    shouldCheckErrors?: boolean,
+    position?: number
+  ) => void;
+  addNonRepeatableComponentToField?: (
+    keys: string[],
+    componentLayoutData: Record<string, unknown>,
+    allComponents: Record<string, unknown>
+  ) => void;
+  addRepeatableComponentToField?: (
+    keys: string[],
+    componentLayoutData: Record<string, unknown>,
+    allComponents: Record<string, unknown>,
+    shouldCheckErrors?: boolean,
+    position?: number
+  ) => void;
+  allLayoutData: {
+    components: Record<string, Schema.Component>;
+    contentType?: Schema.CollectionType | Schema.SingleType;
+  };
+  createActionAllowedFields: string[];
+  formErrors: Record<string, MessageDescriptor>;
+  hasDraftAndPublish: boolean;
+  // TODO: define the tree, can it be inferred?
+  initialData: Record<string, unknown>;
+  isCreatingEntry: boolean;
+  isSingleType: boolean;
+  layout?: Schema.CollectionType | Schema.SingleType;
+  // TODO: define the tree, can it be inferred?
+  modifiedData: Record<string, unknown>;
+  moveComponentDown?: (dynamicZoneName: string, currentIndex: number) => void;
+  moveComponentField?: (payload: { name: string; newIndex: number; currentIndex: number }) => void;
+  moveComponentUp?: (dynamicZoneName: string, currentIndex: number) => void;
+  onChange?: <TAttribute extends Attribute.Any>(
+    payload: {
+      target: { name: string; type?: TAttribute['type']; value: Attribute.GetValue<TAttribute> };
```

Comment:
Sorry sent you in the wrong direction here as I mentioned on discord, type is an input type and not a strapi type. I've got something in place over on https://github.com/strapi/strapi/pull/18045/files that I think maybe makes some sense. If this works for now that's fine we can maybe figure out the difference between the two and what's needed for each after merging.

Reply:
Lets merge this, I'll take a look at your PR and do the changes necessary there 👀